upgrade_test.py

```diff
@@ -1244,3 +1247,166 @@ def _update_argus_upgraded_version(self, node, new_version):
+        Upgrade half of nodes in the cluster, and run rollback.
+        Upgrade all nodes to new version in the end.
+        """
+        cs_user_profiles = self.prepare_data_before_upgrade()
+        # Temporary solution. If we want to have sequential upgrade test, we will need to change the test parameters and build upgrade path
+        new_version = "2022.1"
+        new_scylla_repo = \
+            "https://s3.amazonaws.com/downloads.scylladb.com/deb/ubuntu/scylla-2022.1.list" \
+            "scylla.list"
+        InfoEvent(message=f'Starting rolling upgrade test to {new_version}').publish()
+        self._custom_profile_rolling_upgrade(cs_user_profiles=cs_user_profiles,
+                                             new_version=new_version, new_scylla_repo=new_scylla_repo)
+        InfoEvent(message=f'Finished rolling upgrade test to {new_version}').publish()
+
+        new_version = "2023.1"
+        new_scylla_repo = \
+            "https://s3.amazonaws.com/downloads.scylladb.com/deb/ubuntu/scylla-2023.1.list"
+        InfoEvent(message=f'Starting rolling upgrade test to {new_version}').publish()
+        self._custom_profile_rolling_upgrade(cs_user_profiles=cs_user_profiles,
+                                             new_version=new_version, new_scylla_repo=new_scylla_repo)
+        InfoEvent(message=f'Finished rolling upgrade test to {new_version}').publish()
+
+    def prepare_data_before_upgrade(self):
+        InfoEvent(message='Running a prepare load for the initial custom data').publish()
+        if not (prepare_cs_user_profiles := self.params.get('prepare_cs_user_profiles')):
+            SyntaxError("Parameter 'prepare_cs_user_profiles' is not supplied")
+
+        user_profiles, duration_per_cs_profile = self.parse_cs_user_profiles_param(prepare_cs_user_profiles)
+        stress_before_upgrade = self.run_cs_user_profiles(cs_profiles=user_profiles,
+                                                          duration_per_cs_profile=duration_per_cs_profile)
+        for queue in stress_before_upgrade:
+            self.verify_stress_thread(cs_thread_pool=queue)
+
+        # write workload during entire test
+        InfoEvent(message='Starting write workload during entire test').publish()
+        if not (cs_user_profiles := self.params.get('cs_user_profiles')):
+            SyntaxError("Parameter 'cs_user_profiles' is not supplied")
+
+        return cs_user_profiles
+
+    def _custom_profile_rolling_upgrade(self, cs_user_profiles, new_scylla_repo=None, new_version=None):  # pylint: disable=too-many-locals,too-many-statements
+        user_profiles, duration_per_cs_profile = self.parse_cs_user_profiles_param(cs_user_profiles)
+        entire_write_thread_pool = self.run_cs_user_profiles(cs_profiles=user_profiles,
+                                                             duration_per_cs_profile=duration_per_cs_profile)
+
+        # Let to write_stress_during_entire_test complete the schema changes
```

Comment:
why not using `wait_for_schema_agreement`?

Reply:
cause that what the original rolling upgrade is doing, and more then we are waiting for schema, we are waiting for stress command start writing data

Reply:
changed comment